app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt

```diff
@@ -9,55 +10,73 @@ import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.dialog
 import androidx.navigation.navArgument
+import com.google.accompanist.navigation.material.BottomSheetNavigator
+import com.google.accompanist.navigation.material.ExperimentalMaterialNavigationApi
+import com.google.accompanist.navigation.material.ModalBottomSheetLayout
+import com.google.accompanist.navigation.material.bottomSheet
 import com.intive.picover.articles.view.ArticlesScreen
 import com.intive.picover.images.view.ImagesScreen
 import com.intive.picover.main.navigation.model.NavigationItem
+import com.intive.picover.parties.view.AddPartyBottomSheet
 import com.intive.picover.parties.view.PartiesScreen
 import com.intive.picover.parties.view.PartyDetailsScreen
+import com.intive.picover.parties.viewmodel.PartiesViewModel
 import com.intive.picover.profile.view.DeleteAccountDialog
 import com.intive.picover.profile.view.ProfileScreen
 import com.intive.picover.profile.viewmodel.ProfileViewModel
 
+@OptIn(ExperimentalMaterialNavigationApi::class, ExperimentalMaterial3Api::class)
 @Composable
 fun PicoverNavHost(
 	modifier: Modifier,
 	navController: NavHostController,
+	bottomSheetNavigator: BottomSheetNavigator,
 ) {
-	NavHost(
-		modifier = modifier,
-		navController = navController,
-		startDestination = NavigationItem.PARTIES.route,
-	) {
-		composable("parties") {
-			PartiesScreen(
-				viewModel = hiltViewModel(),
-				navController = navController,
-			)
-		}
-		composable("profile") {
-			ProfileScreen(hiltViewModel(), navController)
-		}
-		dialog("profile/deleteAccount") {
-			val viewModel: ProfileViewModel = hiltViewModel()
-			DeleteAccountDialog(
-				onConfirm = {
-					viewModel.onDeleteAccountClick()
-					navController.popBackStack()
-				},
-				onDismiss = { navController.popBackStack() },
-			)
-		}
-		composable("photos") {
-			ImagesScreen(hiltViewModel())
-		}
-		composable(
-			route = "partyDetails/{partyId}",
-			arguments = listOf(navArgument("partyId") { type = NavType.IntType }),
+	ModalBottomSheetLayout(bottomSheetNavigator = bottomSheetNavigator) {
+		NavHost(
+			modifier = modifier,
+			navController = navController,
+			startDestination = NavigationItem.PARTIES.route,
 		) {
-			PartyDetailsScreen(hiltViewModel())
-		}
-		composable("articles") {
-			ArticlesScreen(hiltViewModel())
+			composable("parties") {
+				val viewModel: PartiesViewModel = hiltViewModel()
+				PartiesScreen(
```

Comment:
minor: can we just use hiltViewModel() in constructor of parties screen, without additional value here ?

Reply:
done